src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,151 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private void validateIndex(int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Your index is <0 or >size.");
+        }
+    }
+
+    private Node<T> getNode(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Your index is <0 or >=size.");
+        }
+        if (index < size / 2) {
+            Node<T> returnNode = head;
+            for (int i = 0; i < index; i++) {
+                returnNode = returnNode.next;
+            }
+            return returnNode;
+        }
+        Node<T> returnNode = tail;
+        for (int i = size - 1; i > index; i--) {
+            returnNode = returnNode.prev;
+        }
+        return returnNode;
+    }
+
+    private void unlink(Node<T> node) {
```

Comment:
According to [Common Mistakes,](https://github.com/mate-academy/jv-linked-list/blob/master/checklist.md) your private methods should be placed after the public ones.